### PR TITLE
Rewrite Host subscription tests without helper class

### PR DESCRIPTION
**Pull request by peterdragun**
This class began really messy and as it was shared there were some tests that had incorrect setup. The main goal was to reduce the complexity of those tests using our shared fixtures

### CODE REVIEW

**Comment by peterdragun** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py -m cli_host_subscription

**Review by pondrejk — APPROVED** [review]

File: tests/foreman/cli/test_host.py
Comment (mshriver) [inline tests/foreman/cli/test_host.py:1866]: `>0` is redundant here, as `assert 0` will fail.

File: pytest_fixtures/component/taxonomy.py
Comment (peterdragun) [reply pytest_fixtures/component/taxonomy.py:70]: Exactly, other fixtures need module_org, that was the reason to create a new fixture.
Before making a new fixture I have looked at the previous one, I don't know the reason behind not reusing the `module_org` in the first place in `module_manifest_org`, but I have found out that this fixture is used a lot and even in the same module as the module_org. So this is just me being safe and not destroying anyone's setup and making their tests fail.

**Comment by mshriver** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py -m cli_host_subscription

File: pytest_fixtures/component/subscription.py
Comment (mshriver) [inline pytest_fixtures/component/subscription.py:15]: I think using `module_default_sat.api.Subscription` and returning an API entity is a better pattern for the fixture, and one that's consistent with the fixture design in robottelo.

File: tests/foreman/cli/test_host.py
Comment (mshriver) [inline tests/foreman/cli/test_host.py:1868]: Using an API entity would allow/force you to make attribute references here instead of key references, like `default_subscription.id`

**Comment by peterdragun** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py -m cli_host_subscription

**Comment by peterdragun** [comment]
All tests failed on installing the katello ca cert :disappointed: There is already raised PR to address those fails: #9705 :partying_face: I think we can wait for that PR to merge and then test this PR again.

**Comment by peterdragun** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py -m cli_host_subscription

**Comment by peterdragun** [comment]
Lint check fail is not related to this PR, same results even after retrying.

File: tests/foreman/cli/test_host.py
Comment (mshriver) [inline tests/foreman/cli/test_host.py:1873]: Feel free to group the tests into classes if you would like to better organize them within the module - or split the module up! Your call - and not in this PR.

**Review by mshriver — APPROVED** [review]

**Pull request merged**
